qa-tests-backend/scripts/run-compatibility.sh

```diff
@@ -20,6 +20,22 @@ source "$ROOT/qa-tests-backend/scripts/lib.sh"
 
 set -euo pipefail
 
+shorten_tag() {
+    if [[ "$#" -ne 1 ]]; then
+        die "Expected a version tag as parameter in shorten_tag: shorten_tag <tag>"
+    fi
+
+    long_tag=$1
+
+    short_tag_regex='([0-9]+\.[0-9]+\.[0-9xX]+)'
+
+    if [[ $long_tag =~ $short_tag_regex ]]; then
+        echo "${BASH_REMATCH[1]}"
+    else
```

Comment:
Looking at that part, I see that you shorten the input by matching an entire rexgexp and returning what was matched. Usually, we shorten things by matching a regexp and returning only parts of what was matched. 

Could you provide few examples of `long_tag` and the respective results of shortening? I would like to understand why this works. This can be done in the PR description as it would document clearly what are we changing here.